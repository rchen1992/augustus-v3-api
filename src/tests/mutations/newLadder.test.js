const { testQuery } = require('@tests/utils');

const userId = '1';
const ladderName = 'My New Ladder';

testQuery({
    id: 'Create new ladder',
    query: `
        mutation createLadder($ladderName: String!) {
            newLadder(ladderName: $ladderName) {
                id
                ladderName
                inviteToken
                userRating
                userRatingDelta
            }
        }
    `,
    variables: {
        userId,
        ladderName,
    },
    context: () => {
        return {
            currentUser: {
                user_id: userId,
            },
            services: {
                ladder: {
                    newLadder() {
                        return {
                            ladder_id: 10,
                            ladder_name: ladderName,
                            invite_token: 'tRJcyp2Jkr13',
                            created_at: '2018-12-11 09:39:31',
                            updated_at: '2018-10-09 01:00:11',
                            ladder_user: {
                                ladder_id: 10,
                                user_id: userId,
                                rating: 1000,
                                rating_delta: 0,
                                created_at: '2018-12-11 09:39:31',
                                updated_at: '2018-10-09 01:00:11',
                            },
                        };
                    },
                },
            },
        };
    },
    expected: undefined,
});
