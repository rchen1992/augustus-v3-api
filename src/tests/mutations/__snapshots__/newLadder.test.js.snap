// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Create new ladder 1`] = `
Object {
  "data": Object {
    "newLadder": Object {
      "id": "10",
      "inviteToken": "tRJcyp2Jkr13",
      "ladderName": "My New Ladder",
      "userRating": 1000,
      "userRatingDelta": 0,
    },
  },
}
`;
